demos/smart_classroom_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,350 @@
+        size_t wait_num_frames = 0;
+        size_t work_num_frames = 0;
+        size_t total_num_frames = 0;
+        size_t work_time_ms_all = 0;
+        const char SPACE_KEY = 32;
+        const char ESC_KEY = 27;
+        bool monitoring_enabled = const_params.actions_type == TOP_K ? false : true;
+        cv::Size graphSize { static_cast<int>(frame_size.width / 4), 60 };
+
+        /** Presenter for rendering system parameters **/
+        Presenter presenter(FLAGS_u, frame_size.height - graphSize.height - 10, graphSize);
+
+        /** Create VideoWriter **/
+        cv::VideoWriter vid_writer;
+        if (!FLAGS_o.empty() && !vid_writer.isOpened()) {
+            vid_writer = cv::VideoWriter(FLAGS_o,
+                                         cv::VideoWriter::fourcc('M', 'J', 'P', 'G'),
+                                         cap.get(cv::CAP_PROP_FPS), frame_size);
+        }
+
+        /** Result containers associated with graph output **/
+        cv::Mat frame, proc, top_k;
+        std::string stream_log, stat_log, det_log;
+        auto out_vector = cv::gout(frame);
+        if (const_params.actions_type == TOP_K) {
+            out_vector += cv::gout(top_k, proc);
+        } else {
+            out_vector += cv::gout(work_num_frames, stream_log, stat_log, det_log, proc);
+        }
+
+        /** TOP_K case starts without processing **/
+        if (const_params.actions_type != TOP_K) cc.start();
+
+        std::cout << "To close the application, press 'CTRL+C' here";
+        if (!FLAGS_no_show) {
+            std::cout << " or switch to the output window and press ESC key";
+        }
+        std::cout << std::endl;
+
+        /** Main cycle **/
+        auto started_all = std::chrono::high_resolution_clock::now();
+        while (true) {
+            auto started = std::chrono::high_resolution_clock::now();
+            char key = cv::waitKey(1);
+            presenter.handleKey(key);
+            if (key == ESC_KEY) {
+                break;
+            }
+            if (const_params.actions_type == TOP_K) {
+                if ((key == SPACE_KEY && !monitoring_enabled) ||
+                    (key == SPACE_KEY && monitoring_enabled)) {
+                    /** SPACE_KEY & monitoring_enabled trigger **/
+                    monitoring_enabled = !monitoring_enabled;
+                    const_params.draw_ptr->ClearTopWindow();
+                }
+            }
+            if (monitoring_enabled) {
+                if (!cc.running()) {
+                    /** TOP_K part. SPACE_KEY is pushed, monitoring enabled
+                     *  Compile and start graph **/
+                    if (!cap.grab()) break;
+                    cc.setSource<custom::CustomCapSource>(cap);
+                    cc.start();
+                }
+                if (!cc.pull(std::move(out_vector))) {
+                    /** Main part. Processing is always on **/
+                    if (cv::waitKey(1) >= 0) break;
```

Comment:
Removed cv::waitKey

Reply:
`/** Main part. Processing is always on **/` here and below seem to be irrelevant

Reply:
This spaghetti demo need more comments. It isn't trivial to understand behavior of main part and TOP_K. I think main demo should get comments about algorithm and processing steps, because GAPI version is reflection of main demo. 

Since all processing placed in graph and this case assumes that it is enabled comment matters. For example TOP_K case starts without processing and simply shows source file and activates processing with SPACE_KEY.

I will expand comments.